src/client/debugger/extension/configuration/resolvers/base.ts

```
@@ -114,13 +114,21 @@ export abstract class BaseConfigurationResolver<T extends DebugConfiguration>
         if (!debugConfiguration) {
             return;
         }
-        if (debugConfiguration.pythonPath === '${command:python.interpreterPath}' || !debugConfiguration.pythonPath) {
+        if (
+            debugConfiguration.python === '${command:python.interpreterPath}' ||
+            debugConfiguration.pythonPath === '${command:python.interpreterPath}' ||
+            (!debugConfiguration.pythonPath && !debugConfiguration.python)
+        ) {
             const pythonPath = this.configurationService.getSettings(workspaceFolder).pythonPath;
             debugConfiguration.pythonPath = pythonPath;
             this.pythonPathSource = PythonPathSource.settingsJson;
         } else {
             this.pythonPathSource = PythonPathSource.launchJson;
         }
+        // We have to ensure 'pythonPath' is always set at the end of this call.
+        if (debugConfiguration.python && !debugConfiguration.pythonPath) {
+            debugConfiguration.pythonPath = debugConfiguration.python;
+        }
```

Why do we, though? If it wasn't set explicitly, it will be set above. If it was set explicitly by the user in their debug configuration - even if that was an empty string or null - we shouldn't silently substitute some other value, but should instead validate (and reject) the value that they have specified.

---

> If it wasn't set explicitly, it will be set above.

That's not the case with the current code actually. The code above doesn't set it if `debugConfig.python` is set.

> Why do we, though?

We currently set `debugConfiguration.debugAdapterPython` and `debugConfiguration.debugLauncherPython` using `debug.pythonPath`. If `pythonPath` is not set at the end of this call, those properties will not be set either.